### DISCUSSION TITLE
Can we generate proof files manually?

### DISCUSSION THREAD

**Discussion by SevenLiquor** [lightninglabs/taproot-assets#1018] [Q&A]
hi,
I realized that I recently had an asset transaction that never arrived, but the on-chain transaction had been confirmed for a long time.I pursued the cause and found that the sender's tapd file directory did not generate the two new proof files for this transaction.The courier universe doesn't have a corresponding leaf either。
The reason for this problem could be caused by an unexpected interruption of tapd, but after restarting tapd didn't try to regenerate this proof file. So I am wondering if there is a way we can manually regenerate the proof file？
thanks~

**Comment by guggero** [comment]
I think it depends on the state of the sender's node and what exactly went wrong. Do you happen to have any logs?
It would theoretically be possible to create a proof manually from the database state. But there is no code for that.

The sender logic should be robust enough to re-attempt the transfer. And maybe the proofs exist as a file on the sender side and it just can't be sent to the universe? Hard to say without any logs, so would appreciate some additional details and log files if available.

**Reply by jharveyb** [reply]
Actually I think the `ExportProof` RPC would try to rebuild the proof from the DB state, if the proof file is not found.
